diff --git a/assets/js/app/rooms/rooms_index.hamlc b/assets/js/app/rooms/rooms_index.hamlc
index 8d700c4..aa09835 100644
--- a/assets/js/app/rooms/rooms_index.hamlc
+++ b/assets/js/app/rooms/rooms_index.hamlc
@@ -1,4 +1,7 @@
 .chat-container__sidebar
   %sidebar
 .chat-container__list-rooms
-  %p TO DO: list rooms here
+  %p TO DO: list rooms BEAUTIFULLY HERE
+  %ul
+    %li(ng-repeat="room in rooms")
+      %a(href="javascript:void(0)" ui-sref="rooms.chat({roomId: room.roomId})") {{room.channelName}}
diff --git a/assets/js/app/routes.js b/assets/js/app/routes.js
index 403b17b..0cf9127 100644
--- a/assets/js/app/routes.js
+++ b/assets/js/app/routes.js
@@ -32,7 +32,24 @@ app.config([
       url: '/rooms',
       views: {
         '': {
-          templateUrl: '/rooms_index.html'
+          templateUrl: '/rooms_index.html',
+          controller: [
+            '$scope',
+            'rooms',
+            ($scope, rooms) => {
+              $scope.rooms = rooms
+            }
+          ],
+          resolve: {
+            rooms: [
+              'RoomsResource',
+              (RoomsResource) => {
+                return RoomsResource.query({room_id: null}).$promise.then((result) => {
+                  return result.rooms
+                })
+              }
+            ]
+          },
         }
       }
     })
diff --git a/assets/js/app/users/users_list.js b/assets/js/app/users/users_list.js
index 0896d3a..f71fdbf 100644
--- a/assets/js/app/users/users_list.js
+++ b/assets/js/app/users/users_list.js
@@ -43,10 +43,6 @@ app.directive('userList', [
             RoomManager.sendFriend(user)
           }
 
-          RoomsResource.query({room_id: null}).$promise.then((result) => {
-            console.log(result)
-          })
-
           controller.setup()
           return controller
         }
